File: 1854-maximum-population-year/1854-maximum-population-year.py
from collections import OrderedDict
class Solution(object):
    def maximumPopulation(self, logs):
        people=[0]*101
        for b,d in logs:
            people[b-1950] += 1
            people[d-1950] -= 1
        for i in range(1,101):
            people[i] += people[i-1]
        maxi=max(people)
        return people.index(maxi)+1950

File: 1854-maximum-population-year/1854-maximum-population-year.java
class Solution 
{
    public int maximumPopulation(int[][] logs) 
    {
        int[] pop = new int[3000];
        for(int[] i:logs)
        {
            pop[i[0]]++;
            pop[i[1]]--;
        }
        int maxi=0;
        int year=1950;
        for(int i=1950;i<=2050;i++)
        {
            pop[i] += pop[i-1];
            if(pop[i]>maxi)
            {
                maxi=pop[i];
                year=i;
            }
        }
        return year;
    }
}